nixos/lib/test-driver/test_driver/driver.py

```diff
@@ -101,6 +106,121 @@ def __exit__(self, *_: Any) -> None:
             for machine in self.machines:
                 machine.release()
 
+    def rebuild(self, cmd: Optional[str] = None, exe: str = sys.argv[0]) -> None:
+        """
+        Only makes sense when running interactively. Rebuilds the test driver by running `cmd`, or a globally defined `driver.rebuild_cmd`. This should be the same command that built this test driver to begin with. Then uses the new driver at path `exe` to reconfigure this one and redeploy changed machines.
+        """
+
+        if cmd is None:
+            cmd = self.rebuild_cmd
+        if cmd is not None:
+            subprocess.run(cmd, shell=True, check=True)
+
+        tmp_dir = get_tmp_dir()
+
+        new_driver_info = subprocess.check_output(
+            [exe, "--internal-print-update-driver-info-and-exit"],
+            text=True,
+        )
+        (
+            start_scripts,
+            vlans_str,
+            testscript,
+            output_directory,
+        ) = new_driver_info.rstrip().split("\n")
```

Comment:
I think it would be better to write these arguments in JSON in the interactive driver derivation output instead. This way we don't have less stuff to run outside of the sandbox, and less guessing about the binary path and its arguments.

You could even add a separate derivation that only outputs this JSON to avoid having to fetch new versions of host packages (qemu) that we won't use here anyway.

Reply:
The reason I did it the way I did is that things seem to be built so that the driver itself is a standalone executable. It gets wrapped in a wrapper that sets arguments. So I tried with this PR to maintain the separation between those layers. But they could also get set by command line arguments or environment variables.

What you're proposing puts us in a state where the executable depends on the wrapper being set up in a particular way, which feels like mixing layers to me.

Maybe I'm being overly concerned about this though, and the layers are already mixed up or we just don't care if we mix them up. If so, I think your solution is indeed better, but I didn't know enough to make that judgment call myself.